zipline/utils/factory.py

```diff
@@ -224,65 +224,60 @@ def create_returns_from_list(returns, sim_params):
                      data=returns)
 
 
-def create_daily_trade_source(sids, trade_count, sim_params,
-                              concurrent=False):
+def create_daily_trade_source(sids, sim_params, concurrent=False):
     """
     creates trade_count trades for each sid in sids list.
     first trade will be on sim_params.period_start, and daily
     thereafter for each sid. Thus, two sids should result in two trades per
     day.
-
-    Important side-effect: sim_params.period_end will be modified
-    to match the day of the final trade.
     """
     return create_trade_source(
         sids,
-        trade_count,
         timedelta(days=1),
         sim_params,
         concurrent=concurrent
     )
 
 
-def create_minutely_trade_source(sids, trade_count, sim_params,
-                                 concurrent=False):
+def create_minutely_trade_source(sids, sim_params, concurrent=False):
     """
     creates trade_count trades for each sid in sids list.
     first trade will be on sim_params.period_start, and every minute
     thereafter for each sid. Thus, two sids should result in two trades per
     minute.
-
-    Important side-effect: sim_params.period_end will be modified
-    to match the day of the final trade.
     """
     return create_trade_source(
         sids,
-        trade_count,
         timedelta(minutes=1),
         sim_params,
         concurrent=concurrent
     )
 
 
-def create_trade_source(sids, trade_count,
-                        trade_time_increment, sim_params,
+def create_trade_source(sids, trade_time_increment, sim_params,
                         concurrent=False):
 
+
+    # If the sim_params define an end that is during market hours, that will be
+    # used as the end of the data source
+    if trading.environment.is_market_hours(sim_params.period_end):
+        end = sim_params.period_end
+    # Otherwise, the last_close after the period_end is used as the end of the
+    # data source
+    else:
+        end = sim_params.last_close
```

Comment:
@ehebert -- Does this logic make sense? This allows us to define period_ends that are during a day, but if a time outside of the trading day is given, the last_close of that day is used.